loader/common/log.c

```diff
@@ -13,6 +14,63 @@ enum log_level logger_set_level(enum log_level level)
     return prev;
 }
 
+#ifdef HYPER_SERIAL_LOG
+#define SERIAL_COM1 0x3F8
```

Comment:
I was thinking maybe we could improve the #ifdef hell here if we just wrap the entire thing in one ifdef like so
```c
#ifdef HYPER_SERIAL_LOG
// actual implementation
#else
static void serial_init(void) {}

static void write_serial(const char *msg, size_t len)
{
    UNUSED(msg);
    UNUSED(len);
}
#endif
```